projects/power_distribution/test/test_front_uv_detector.c

```diff
@@ -0,0 +1,55 @@
+#include "can_transmit.h"
+#include "front_uv_detector.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "ms_test_helpers.h"
+#include "pd_events.h"
+
+static GpioAddress s_uv_comp_pin_address = { .port = GPIO_PORT_B, .pin = 0 };
+
+static volatile bool s_interrupt_ran = false;
+
+static CanStorage s_can_storage;
+
+static void prv_initialize_can(SystemCanDevice can_device) {
+  CanSettings can_settings = {
+    .device_id = can_device,
+    .loopback = true,
+    .bitrate = CAN_HW_BITRATE_500KBPS,
+    .rx_event = POWER_DISTRIBUTION_CAN_EVENT_RX,
+    .tx_event = POWER_DISTRIBUTION_CAN_EVENT_TX,
+    .fault_event = POWER_DISTRIBUTION_CAN_EVENT_FAULT,
+    .tx = { GPIO_PORT_A, 12 },
+    .rx = { GPIO_PORT_A, 11 },
+  };
+  can_init(&s_can_storage, &can_settings);
+}
```

Comment:
Since the CAN device is constant (it's always `SYSTEM_CAN_DEVICE_POWER_DISTRIBUTION_FRONT` since this module is always on front PD), you don't need this function, which is used to initialize CAN with a different CAN device per test - you can just call `initialize_can_and_dependencies` from ms_test_helper_can.h in `setup_test`, which will also initialize gpio, interrupts, soft timers, and the event queue for you.